src/library/scala/collection/View.scala

```diff
@@ -163,6 +163,64 @@ object View extends IterableFactory[View] {
     def iterator = partition.underlying.iterator.filter(x => partition.p(x) == cond)
   }
 
+  /** A view that splits an underlying collection into two views */
+  class MapSplit[A, A1, A2](val underlying: SomeIterableOps[A], val f: A => Either[A1, A2]) {
+
+    /** The view consisting of all elements of the underlying collection
+      *  that map to `Left`.
+      */
+    val left: View[A1] = new AbstractView[A1] {
```

Comment:
`left` and `right` should be named classes that define a `@SerialVersionUID` (see `Partition` for an example)